tests/src/Simple/Threading/Threading.cs

```diff
@@ -391,6 +394,25 @@ public static void SemaphoreTest()
     }
 
     [Fact]
+    public static void Reproduce3616()
```

Comment:
Should call this test something that reflects what it tests for. A full link to the issue in a comment would be useful.